presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -129,7 +129,7 @@ public PlanNode optimize(PlanNode plan, Session session, TypeProvider types, Pla
             return plan;
         }
 
-        rulesTriggered.stream().forEach(x -> session.getOptimizerInformationCollector().addInformation(new PlanOptimizerInformation(x, true, Optional.empty())));
+        rulesTriggered.stream().forEach(x -> session.getOptimizerInformationCollector().addInformation(new PlanOptimizerInformation(x, true, Optional.empty()), session.getQueryId()));
```

Comment:
the concurrent modification problem is with rulesTriggered, not the OptimizationInformationCollector.